Sources/FunctionalTableData/CellActions.swift

```diff
@@ -117,6 +117,105 @@ public struct CellActions {
+				self.style = style
+				self.image = image
+			}
+			
+			@available(iOS 13.0, *)
+			fileprivate func asUIMenuElement() -> UIMenuElement {
+				let style: UIMenuElement.Attributes
+				switch self.style {
+				case .disabled:
+					style = .disabled
+				case .destructive:
+					style = .destructive
+				default:
+					style = UIMenuElement.Attributes()
+				}
+				
+				switch self.type {
+				case .action(let action):
+					return UIAction(title: title, image: image, identifier: nil, attributes: style, handler: { _ in action() } )
+				case .submenuParent(let inlined, let children):
+					let options = inlined ? UIMenu.Options.displayInline : []
+					let submenuChildren = children.map { $0.asUIMenuElement() }
+					return UIMenu(title: title, image: image, identifier: nil, options: options, children: submenuChildren)
+				}
+			}
+		}
+		
+		/// Closure type that is executed when the system needs a view controller to preview with the context menu.
+		///
+		/// __Note:__ If this value is nil or returns nil, the system will default to using the entire cell as a preview.
+		public typealias PreviewContentProvider = () -> UIViewController?
+		/// Closure type that is executed when the user taps on the context menu preview.
+		/// - parameter previewingViewController: The view controller currently being previewed. If the `PreviewContentProvider` returns nil, this parameter will be nil.
+		public typealias PreviewContentCommitter = (_ previewingViewController: UIViewController?) -> Void
+		/// Closure type that is executed  when the system needs the elements for the context menu.
+		public typealias MenuActionsProvider = () -> [MenuItem]
+		
+		let title: String
+		let previewContentProvider: PreviewContentProvider?
+		let previewContentCommitter: PreviewContentCommitter?
+		let menuActionsProvider: MenuActionsProvider
+		
+		/// Creates a context menu configuration object with the specified set of actions.
+		///
+		/// - Parameters:
+		///   - title: The title of the context menu. Defaults to `""`.
+		///   - previewContentProvider: Closure that provides an custom preview view controller for the context menu. Defaults to `nil`.
+		///   - previewContentComitter: Closure that is executed when the user taps the context menu preview and should navigate to that screen. Defaults to `nil`.
+		///   - menuActionsProvider: Closure that is executed when the system needs the actions to use in the context menu.
+		public init(title: String = "", preivewContentProvider: PreviewContentProvider? = nil, previewContentCommitter: PreviewContentCommitter? = nil, menuActionsProvider: @escaping MenuActionsProvider) {
+			self.title = title
+			self.previewContentProvider = preivewContentProvider
+			self.previewContentCommitter = previewContentCommitter
+			self.menuActionsProvider = menuActionsProvider
+		}
+		
+		@available(iOS 13.0, *)
+		internal func asUIContextMenuConfiguration(with identifier: IndexPath? = nil) -> UIContextMenuConfiguration {
```

Comment:
I think the `ItemPath` would be preferred over the `IndexPath` for a couple reasons:
- it would always be non-nil
- it should be relatively static, meaning that if an item moves from one index path to another (but is still the same thing) then it's ItemPath doesn't change (only if it moves sections)

Reply:
This is great context I didn't have. Thank you!